src/libutil/filesystem.cpp

```diff
@@ -158,42 +158,56 @@ Filesystem::searchpath_split (const std::string &searchpath,
 
 
 std::string
-Filesystem::searchpath_find (const std::string &filename,
+Filesystem::searchpath_find (const std::string &filename_utf8,
                              const std::vector<std::string> &dirs,
                              bool testcwd, bool recursive)
 {
-    bool abs = Filesystem::path_is_absolute (filename);
+#ifdef _WIN32
+    const boost::filesystem::path filename (Strutil::utf8_to_utf16 (filename_utf8));
+#else
+    const boost::filesystem::path filename (filename_utf8);
+#endif
+    bool abs = filename.is_absolute();
 
     // If it's an absolute filename, or if we want to check "." first,
     // then start by checking filename outright.
     if (testcwd || abs) {
-        if (Filesystem::is_regular (filename))
-            return filename;
+        if (boost::filesystem::is_regular_file (filename))
+            return filename_utf8;
     }
 
     // Relative filename, not yet found -- try each directory in turn
-    BOOST_FOREACH (const std::string &d, dirs) {
+    BOOST_FOREACH (const std::string &d_utf8, dirs) {
         // std::cerr << "\tPath = '" << d << "'\n";
-        boost::filesystem::path f = d;
-        f /= filename;
```

Comment:
Assigning a string to a `path` was triggering a conversion to wide characters using the boost default conversion facet.